cluster-operator/src/main/java/io/strimzi/operator/cluster/model/cruisecontrol/CpuCapacity.java

```diff
@@ -4,41 +4,118 @@
+     */
+    public CpuCapacity(BrokerCapacity brokerCapacity,
+                        BrokerCapacityOverride brokerCapacityOverride,
+                        ResourceRequirements resourceRequirements) {
+        this(CPU.processResourceCapacity(brokerCapacity, brokerCapacityOverride, resourceRequirements));
     }
 
-    protected static String milliCpuToCpu(int milliCPU) {
+    private static String milliCpuToCpu(int milliCPU) {
         return String.valueOf(milliCPU / 1000.0);
     }
 
     /**
-     * Returns CpuCapacity object as a JsonObject
+     * Checks whether all Kafka broker pods have their CPU resource requests equal to their CPU limits.
+     *
+     * @param kafkaBrokerResources a map of broker pod names to their {@link ResourceRequirements}
+     * @return {@code true} if all brokers have matching CPU requests and limits; {@code false} otherwise
+     */
+    public static boolean cpuRequestsMatchLimits(Map<String, ResourceRequirements> kafkaBrokerResources) {
+        if (kafkaBrokerResources == null) {
+            return false;
+        }
+        for (ResourceRequirements resourceRequirements : kafkaBrokerResources.values()) {
+            Quantity request = ResourceRequirementsUtils.getCpuRequest(resourceRequirements);
+            Quantity limit = ResourceRequirementsUtils.getCpuLimit(resourceRequirements);
+            if (request == null || limit == null || request.compareTo(limit) != 0) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    /**
+     * Derives the CPU capacity from Kubernetes {@link ResourceRequirements}.
+     * <p>
+     * This method first attempts to extract the CPU request value from the resource requirements.
+     * If the request is defined, it is converted from millicores to cores and returned as a {@link String}.
+     * If no request is present, it falls back to the CPU limit (if available) and performs the same conversion.
+     * If neither a request nor a limit is specified, {@code null} is returned.
+     *
+     * @param resourceRequirements The Kubernetes resource requirements containing CPU requests and/or limits.
+     * @return The CPU capacity in cores as a {@link String}, or {@code null} if no CPU values are defined.
+     */
+    public static String getCpuBasedOnRequirements(ResourceRequirements resourceRequirements) {
+        Quantity request = ResourceRequirementsUtils.getCpuRequest(resourceRequirements);
+        Quantity limit = ResourceRequirementsUtils.getCpuLimit(resourceRequirements);
+
+        if (request != null) {
+            int milliCpus = Quantities.parseCpuAsMilliCpus(request.toString());
+            return CpuCapacity.milliCpuToCpu(milliCpus);
+        } else if (limit != null) {
+            int milliCpus = Quantities.parseCpuAsMilliCpus(limit.toString());
+            return CpuCapacity.milliCpuToCpu(milliCpus);
+        } else {
+            return null;
+        }
```

Comment:
Why isn't this whole thing in your Utils class together with the other use for this method?

Reply:
You are right, it would make more sense there, just moved it to the Utils class in latest commit.